request_information.go

```diff
@@ -1,566 +1,636 @@
+func NewRequestInformationWithMethodAndUrlTemplateAndPathParameters(method HttpMethod, urlTemplate string, pathParameters map[string]string) *RequestInformation {
+	value := NewRequestInformation()
+	value.Method = method
+	value.UrlTemplate = urlTemplate
+	value.PathParameters = pathParameters
+	return value
+}
+func ConfigureRequestInformation[T any](request *RequestInformation, config *RequestConfiguration[T]) {
+	if request == nil {
+		return
+	}
+	if config == nil {
+		return
+	}
+	if config.QueryParameters != nil {
+		request.AddQueryParameters(*(config.QueryParameters))
+	}
+	request.Headers.AddAll(config.Headers)
+	request.AddRequestOptions(config.Options)
+}
+
+// GetUri returns the URI of the request.
+func (request *RequestInformation) GetUri() (*u.URL, error) {
+	if request.uri != nil {
+		return request.uri, nil
+	} else if request.UrlTemplate == "" {
+		return nil, errors.New("uri cannot be empty")
+	} else if request.PathParameters == nil {
+		return nil, errors.New("uri template parameters cannot be nil")
+	} else if request.QueryParameters == nil {
+		return nil, errors.New("uri query parameters cannot be nil")
+	} else if request.QueryParametersAny == nil {
+		return nil, errors.New("uri query parameters cannot be nil")
+	} else if request.PathParameters[raw_url_key] != "" {
+		uri, err := u.Parse(request.PathParameters[raw_url_key])
+		if err != nil {
+			return nil, err
+		}
+		request.SetUri(*uri)
+		return request.uri, nil
+	} else {
+		_, baseurlExists := request.PathParameters["baseurl"]
+		if !baseurlExists && strings.Contains(strings.ToLower(request.UrlTemplate), "{+baseurl}") {
+			return nil, errors.New("pathParameters must contain a value for \"baseurl\" for the url to be built")
+		}
+
+		substitutions := make(map[string]any)
+		for key, value := range request.PathParameters {
+			substitutions[key] = request.sanitizeValue(value)
+		}
+		for key, value := range request.PathParametersAny {
+			substitutions[key] = request.normalizeParameters(reflect.ValueOf(value), request.sanitizeValue(value), false)
+		}
+		for key, value := range request.QueryParameters {
+			substitutions[key] = request.sanitizeValue(value)
+		}
+		for key, value := range request.QueryParametersAny {
+			substitutions[key] = request.sanitizeValue(value)
+		}
+		url, err := stduritemplate.Expand(request.UrlTemplate, substitutions)
+		if err != nil {
+			return nil, err
+		}
+		uri, err := u.Parse(url)
+		return uri, err
+	}
+}
+
+func castItem[T any, R interface{}](collection []T, mutator func(t T) R) []R {
+	if len(collection) > 0 {
+		cast := make([]R, len(collection))
+		for i, v := range collection {
+			cast[i] = mutator(v)
+		}
+		return cast
+	}
+	return nil
+}
+
+func (request *RequestInformation) sanitizeValue(value any) any {
+	if value == nil {
+		return nil
+	}
+
+	switch v := value.(type) {
+	case *time.Time:
```

Comment:
Out of curiosity, do we expect both `*time.Time` and `time.Time` instances from generation or are we doing this to just be sure?

Reply:
In this case it's defensive programming, for any possible edge case